app/models/lettings_log.rb

```diff
@@ -53,6 +53,15 @@ class LettingsLog < Log
   scope :filter_by_before_startdate, ->(date) { where("lettings_logs.startdate >= ?", date) }
   scope :unresolved, -> { where(unresolved: true) }
   scope :filter_by_organisation, ->(org, _user = nil) { where(owning_organisation: org).or(where(managing_organisation: org)) }
+  scope :duplicate_logs, lambda { |log|
+                           attrs_to_check = %w[tenancycode propcode startdate age1 sex1 ecstat1 tcharge postcode_full]
+                           where(log.slice(*attrs_to_check))
+                           .where.not(id: log.id)
+                           .where.not(status: "deleted")
+                           .where.not("startdate IS NULL OR age1 IS NULL OR sex1 IS NULL OR ecstat1 IS NULL OR tcharge IS NULL OR postcode_full IS NULL OR propcode IS NULL OR needstype IS NULL")
+                           .where("location_id = ? OR needstype = 1", log.location_id)
+                         }
```

Comment:
since this is over many lines now I'm surprised rubocop doesn't enforce do end rather than {} wouldn't that be preferable as standard ruby for a multiline block that is not having something chained on the end of it?

I'm inclined to want to pull attrs_to_check into a constant on the LettingsLog class, hopefully there is a way then to access this list from the various places where it is used, eg bulk upload

Reply:
This is what rubocop corrects it to, I think we need to use lambda cause it's multiline and this is the syntax for it.

Good should, extracted the constants 👍